src/chainparams.cpp

```diff
@@ -777,29 +777,29 @@ class CRegTestParams : public CChainParams {
 };
 static CRegTestParams regTestParams;
 
-static CChainParams *pCurrentParams = 0;
+static const CChainParams* pCurrentParams = 0;
 
-const CChainParams &Params() {
+const CChainParams& Params() {
     assert(pCurrentParams);
     return *pCurrentParams;
 }
 
-CChainParams& Params(const std::string& chain)
+const CChainParams* Params(const std::string& chain)
```

Comment:
Why does this change to a pointer?

Reply:
There isn't actually a way to have this return a const reference, because the value that it returns may be a reference derived from one of three different pointers. However, those pointers are all const, so returning this as a const pointer (i.e. a pointer that preserves the const-ness of its pointee) is fine, so I judged that to be better than the current situation.

Reply:
Nope, I was wrong, I must have made some other silly error. Fixed.